Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp

```diff
@@ -643,7 +663,8 @@ class CombinatorialKalmanFilter {
         }
         // Update state and stepper with post material effects
         materialInteractor(surface, state, stepper, postUpdate);
-      } else if (surface->surfaceMaterial() != nullptr) {
+      } else if (surface->associatedDetectorElement() != nullptr ||
+                 surface->surfaceMaterial() != nullptr) {
```

Comment:
That would count holes, right?

Reply:
Yes this correspond to material surface and holes. (It also use to count for both but that meant holes could only be found on surface with material)